sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -83,6 +83,16 @@ case class SerializeFromObject(
     child: LogicalPlan) extends UnaryNode with ObjectConsumer {
 
   override def output: Seq[Attribute] = serializer.map(_.toAttribute)
+
+  // We can't estimate the size of ObjectType. We implement statistics here to avoid
+  // directly estimate any child plan which produces domain objects as output.
+  override def statistics: Statistics = {
+    if (child.output.find(_.dataType.isInstanceOf[ObjectType]).isDefined) {
```

Comment:
nit: we can just do `child.output.head.isInstanceOf[ObjectType]`, this is guarantted in:
```
trait ObjectConsumer extends UnaryNode {
   assert(child.output.length == 1)
   ...
```